### PR TITLE
Disallow type predicate as type assertion

### LINKED ISSUE #5731: Type-guard-assertion-like syntax does not do anything
The following code does not cause any error nor does type-guard thing.

```typescript
let foo: Node
if (<foo is Element>(foo.nodeType === 1)) {	
}
```

Is this by design? The language service implicitly thinks the `foo` inside type assertion as `any`. I would prefer to making this as a valid type guard syntax if there is no spec about this.

### PR DESCRIPTION
**Pull request by sandersn**
Fixes #5731 by disallowing the construction for now.

### CODE REVIEW

**Comment by sandersn** [comment]
@weswigham @ahejlsberg could you take a look since you commented on the original issue.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -9656,16 +9656,22 @@ namespace ts {
         function checkAssertion(node: AssertionExpression) {
             const exprType = getRegularTypeOfObjectLiteral(checkExpression(node.expression));
             const targetType = getTypeFromTypeNode(node.type);
-            if (produceDiagnostics && targetType !== unknownType) {
-                const widenedType = getWidenedType(exprType);
+            if (produceDiagnostics) {
+                if (node.type.kind === SyntaxKind.TypePredicate) {
+                    error(node.type, Diagnostics.A_type_predicate_is_not_allowed_as_part_of_a_type_assertion_expression);
```
Comment (weswigham) [inline src/compiler/checker.ts]: Rather than creating a new error message, why not call `checkTypePredicate`, which aught to error here?

Comment (sandersn) [inline src/compiler/checker.ts]: Thanks, I didn't know about that.

**Comment by sandersn** [comment]
OK, the change is much smaller now.

**Pull request closed without merging**